### ISSUE TITLE
Memory exhaustion with Script Console and big scripts

### ISSUE THREAD

**Issue by thc202** [zaproxy/zaproxy#2675]
Issue reported in the IRC channel.

Steps to reproduce the issue:
1. Run ZAP with Script Console and Zest add-ons installed and with the option "JVM" > "JVM Options" set to "-Xmx64m" (easier and faster to reproduce the issue);
2. Start recording a Stand Alone Zest script;
3. Proxy a lot of requests (the exact number depends on the size of the responses, but more than 25 should trigger the issue, Zest script should be around 900 lines);
4. Note that ZAP becomes unresponsive (e.g. GUI freezes for some moments or permanently);
5. Check the log file, there should be some exceptions, for example:
```
141699 [AWT-EventQueue-0] ERROR org.zaproxy.zap.ZAP$UncaughtExceptionLogger  - Exception in thread "AWT-EventQueue-0"
java.lang.OutOfMemoryError: Java heap space
	at javax.swing.text.GapContent.getPositionsInRange(GapContent.java:744)
	at javax.swing.text.GapContent$RemoveUndo.<init>(GapContent.java:914)
	at javax.swing.text.GapContent.remove(GapContent.java:150)
	at javax.swing.text.AbstractDocument.handleRemove(AbstractDocument.java:615)
	at javax.swing.text.AbstractDocument.remove(AbstractDocument.java:590)
	at javax.swing.text.AbstractDocument.replace(AbstractDocument.java:666)
	at javax.swing.text.JTextComponent.setText(JTextComponent.java:1669)
	at org.zaproxy.zap.extension.scripts.CommandPanel.clear(Unknown Source)
	at org.zaproxy.zap.extension.scripts.ConsolePanel.updateCommandPanelState(Unknown Source)
	at org.zaproxy.zap.extension.scripts.ConsolePanel.setScript(Unknown Source)
	at org.zaproxy.zap.extension.scripts.ExtensionScriptsUI.displayScript(Unknown Source)
	at org.zaproxy.zap.extension.zest.ExtensionZest.display(Unknown Source)
	at org.zaproxy.zap.extension.zest.ExtensionZest.addToParent(Unknown Source)
	at org.zaproxy.zap.extension.zest.ExtensionZest$4.run(Unknown Source)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:756)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:726)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

ZAP:
Version 2.5.0.

Add-on:
Script Console version 16 (older versions are also affected).

**Comment by thc202** [comment]
Fixed in zaproxy/zap-extensions#457.

**Comment by thc202** [comment]
Released in version 17 of Script Console add-on.
